### PR TITLE
feat(pds-ember): add DefinitionList component

### PR DESCRIPTION
**Pull request by CITguy**
Port component from HCP:cloud-ui-core addon.

JIRA:
* PDS-54
* PDS-98
* PDS-134

### Screenshots
_mobile (stacked)_
mobile - stacked

_desktop (grid)_
desktop - grid

_desktop (stacked)_
desktop - stacked

### CODE REVIEW

File: packages/pds-ember/addon/components/pds/definition-list/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Component from '@glimmer/component'
+
+/**
+ * @class PdsDefinitionList
+ */
+
+/**
+ * When true, forces the list to render a stacked layout.
+ *
+ * @argument stacked
```
Comment (meirish) [inline packages/pds-ember/addon/components/pds/definition-list/index.js:10]: Is there a default breakpoint this happens at, or is it entirely controlled by the arg?

File: packages/pds-ember/app/styles/pds/components/definition-list/__config.scss [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+@use "../../tokens/scale";
+@use "../../utils/media" as Media;
+@use "../_config" as Component;
+
+// atomic vars
+$color: color.$ui-gray-900;
+$columnGap: theme.$size--lg; // 1.5rem (24px)
+$rowGap: scale.$lg--1; // 1.25rem (20px)
+// composed vars
+$gridGap: $rowGap $columnGap;
```
Comment (meirish) [inline packages/pds-ember/app/styles/pds/components/definition-list/__config.scss:19]: Are these going to be shared more widely or do you think they'll be specific to this component?

**Review by meirish — APPROVED** [review]
Looks good! I was curious about some things, but don't think it needs changed.

File: packages/pds-ember/addon/components/pds/definition-list/index.js (hunk 1 above)
Comment (CITguy) [reply packages/pds-ember/addon/components/pds/definition-list/index.js:10]: This is an argument that was brought in to allow application of [styles present in HCP](https://github.com/hashicorp/cloud-ui/blob/master/packages/cloud-ui-core/addon/styles/components/definition-list/index.scss#L97-L99).

* Mobile: 
    * the list is always stacked
* Desktop:
    * if browser doesn't support grid layout, it falls back to being stacked
    * if browser does support grid layout, it applies a grid layout unless overridden by `@stacked={{true}}`

File: packages/pds-ember/app/styles/pds/components/definition-list/__config.scss (hunk 2 above)
Comment (CITguy) [reply packages/pds-ember/app/styles/pds/components/definition-list/__config.scss:19]: not sure yet, but that's why it's defined in a private module :)

**Pull request merged**